packages/core/src/schema.ts

```diff
@@ -0,0 +1,3 @@
+export function createSchema<TSchema>(): TSchema {
+  return (null as any) as TSchema;
```

Comment:
It would be more true if this would return `undefined` as this lands, for example, on the `.schema.states` which has a type of `T | undefined`. 

It still would not be type-safe at runtime though as this function returns `TSchema` and not `TSchema | undefined`. I've implemented the same thing as part of the https://github.com/davidkpiano/xstate/pull/1543 and I have used a slightly other direction there. "Schemas" are functions accepting a particular argument, see here: https://github.com/davidkpiano/xstate/blob/ab4d9c165d84de7d0669a012950dfe0acbe2dad5/packages/core/src/schemas.ts which makes it possible to leverage inference in a type-safe manner, see the types: https://github.com/davidkpiano/xstate/blob/ab4d9c165d84de7d0669a012950dfe0acbe2dad5/packages/core/src/types.ts#L761-L764

My approach doesn't allow for accessing the schema at runtime though and I suppose that we are aiming for that to facilitate better visualization~. But this one doesn't really allow for that either as this always returns `null`. Could we try to account for this use case as it seems to be the one that you very much like to see land in the end?

Maybe something like this:
```ts
export function createSchema<TSchema>(jsonSchema: any): (item: TSchema) => any  {
  const validator = (item: TSchema) => {}
  validator.$schema = jsonSchema
  return validator
}
```

From what I understand there is also a plan to opt-in to some validations in the future. Do you have an idea how this would work? How one would opt-into them and what would happen on a failed validation?

Reply:
Added an example to c2b7e15 - in my view, a schema is not a validation function, but rather one that can be used for validation. It should be completely serializable:

```js
// ...
const machine = createMachine({
  schema: {
    events: createSchema<SomeEvents>({
      anyOf: [
        { type: 'object', properties: {/* ... */} },
        { type: 'object', properties: {/* ... */} },
        // ...
      ]
    })
  }
});
```

In the above example, JSON Schema for the events is passed in (this doesn't have to be JSON Schema, it can be anything, including a function), and this is saved on `machine.schema.events` as-is. (Ideally for TS, we'd use opaque/nominal/??? types for this, since what's passed in and returned is obviously not `SomeEvents` but it _represents_ `SomeEvents`)

Then, it can be used in whichever way the developer wants, for example:

```js
// using json-schema-faker
function sendSampleEvent(service) {
  const sampleEvent = jsf.generate(service.machine.schemas.events);

  service.send(sampleEvent);
}
```

Reply:
Ok, let's roll with that - if you think it's worth it then we could use opaque types for this but I also don't care that much in the end.